frontend/src/components/MinioArtifactPreview.test.tsx

```diff
@@ -0,0 +1,139 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { render, act } from '@testing-library/react';
+import MinioArtifactPreview from './MinioArtifactPreview';
+import { Apis } from '../lib/Apis';
+
+jest.mock('../lib/Apis');
+
+describe('MinioArtifactPreview', () => {
+  const readFile = Apis.readFile as jest.Mock;
+
+  beforeEach(() => {
+    readFile.mockResolvedValue('preview ...');
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('handles undefined artifact', () => {
+    const { container } = render(<MinioArtifactPreview artifact={undefined} />);
+    expect(container).toMatchInlineSnapshot(`<div />`);
+  });
+
+  it('handles null artifact', () => {
+    const { container } = render(<MinioArtifactPreview artifact={null as any} />);
+    expect(container).toMatchInlineSnapshot(`<div />`);
+  });
+
+  it('handles empty artifact', () => {
+    const { container } = render(<MinioArtifactPreview artifact={{} as any} />);
+    expect(container).toMatchInlineSnapshot(`<div />`);
+  });
+
+  it('handles invalid artifact: no bucket', () => {
+    const s3Artifact = {
+      accessKeySecret: { key: 'accesskey', optional: false, name: 'minio' },
+      bucket: '',
+      endpoint: 'minio.kubeflow',
+      key: 'bar',
+      secretKeySecret: { key: 'secretkey', optional: false, name: 'minio' },
+    };
+    const { container } = render(<MinioArtifactPreview artifact={s3Artifact} />);
+    expect(container).toMatchInlineSnapshot(`<div />`);
+  });
+
+  it('handles invalid artifact: no key', () => {
+    const s3Artifact = {
+      accessKeySecret: { key: 'accesskey', optional: false, name: 'minio' },
+      bucket: 'foo',
+      endpoint: 'minio.kubeflow',
+      key: '',
+      secretKeySecret: { key: 'secretkey', optional: false, name: 'minio' },
+    };
+    const { container } = render(<MinioArtifactPreview artifact={s3Artifact} />);
+    expect(container).toMatchInlineSnapshot(`<div />`);
+  });
+
+  it('handles s3 artifact', async () => {
+    const s3Artifact = {
+      accessKeySecret: { key: 'accesskey', optional: false, name: 'minio' },
+      bucket: 'foo',
+      endpoint: 's3.amazonaws.com',
+      key: 'bar',
+      secretKeySecret: { key: 'secretkey', optional: false, name: 'minio' },
+    };
+
+    const container = document.body.appendChild(document.createElement('div'));
+    await act(async () => {
+      render(<MinioArtifactPreview artifact={s3Artifact} />, { container });
+    });
```

Comment:
This is kind of strange. Why did you need to create the div manually and wrap `render` in an `act`?

Reply:
Did you want to try something like https://github.com/kubeflow/pipelines/blob/f882f361cc705580261128e33523fe80d781b30f/frontend/src/components/PodYaml.test.tsx#L37? to wait for promise to resolve before finally rendering the container?